homeassistant/components/http/cors.py

```diff
@@ -79,12 +79,7 @@ def _allow_cors(
         },
     )
 
-    if not origins:
-        return
-
-    async def cors_startup(app: Application) -> None:
-        """Initialize CORS when app starts up."""
-        for resource in list(app.router.resources()):
-            _allow_cors(resource)
-
-    app.on_startup.append(cors_startup)
+    if origins:
+        app["allow_configured_cors"] = _allow_cors
+    else:
+        app["allow_configured_cors"] = lambda _: None
```

Comment:
Can you add a test to validate this behavior now works as expected

Reply:
sorry, saw the comment to late :/